### PR TITLE
install dashboard on minishift/openshift

### PR DESCRIPTION
**Pull request by Gl4di4torRr**
# Changes
Adding service, route, image stream and build config to simplify the process to install the dashboard.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ y] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: config/tekton-dashboard-deployment.yaml [hunk 1]
```diff
@@ -1,42 +1,120 @@
-apiVersion: apps/v1
-kind: Deployment
+apiVersion: v1
+kind: Template
```
Comment (mnuttall) [inline config/tekton-dashboard-deployment.yaml]: Hi @Gl4di4torRr this file is currently being used as a Deployment for those applying the changes directly to their local environment, such as Docker Desktop, minikube etc. I'm concerned that changing this from `kind: Deployment` to `kind: Template` is going to break a number of existing users. Would it be possible to provide the Template as a separate config element, perhaps in its own file?

Comment (Gl4di4torRr) [reply config/tekton-dashboard-deployment.yaml]: @mnuttall maybe there should be a split between defining specific object types and templates? I can created a folder `config/templates` and place my template files there and leave the deployment object alone for the sake of existing users.

File: config/tekton-dashboard.yaml [hunk 2]
```diff
@@ -0,0 +1,59 @@
+# ------------------- Dashboard Deployment ------------------- #
```
Comment (mnuttall) [inline config/tekton-dashboard.yaml]: I'm being a bit fussy now but the Deployment section of this file shouldn't be showing any changes. Is this just git getting confused or have you moved and recreated the file?

File: config/tekton-dashboard-deployment.yaml (hunk 1 above)
Comment (mnuttall) [reply config/tekton-dashboard-deployment.yaml]: Thank you!

File: config/tekton-dashboard.yaml (hunk 2 above)
Comment (mnuttall) [reply config/tekton-dashboard.yaml]: Maybe I had an old copy of this - it's looking much cleaner now.

**Comment by mnuttall** [comment]
I think we're good to go once you've got your conflicts resolved and changes squashed down to a single commit ... because Tekton uses Prow, we don't get the glorious 'squash and commit' button, and have to do it ourselves : (

**Comment by Gl4di4torRr** [comment]
@mnuttall okay, i think i got everything cleaned up and squashed the way you want it.
Interesting fact about Prow - i did not know that. Was wondering why you were asking me to squash commits manually :)

**Comment by mnuttall** [comment]
/approve
/lgtm
Thank you!

**Pull request merged**